scene/gui/line_edit.cpp

```diff
@@ -1210,7 +1210,12 @@ void LineEdit::delete_text(int p_from_column, int p_to_column) {
 
 void LineEdit::set_text(String p_text) {
 	clear_internal();
-	append_at_cursor(p_text);
+
+	if (max_length <= 0 || max_length > p_text.length()) {
+		append_at_cursor(p_text);
+	} else {
+		append_at_cursor(p_text.substr(0, max_length));
```

Comment:
This is not correct, as we're *appending* and `max_length` governs the max length of the final string (not the one being pasted). So it should restrict to `max_length - text.length()`.

I'll make another PR doing the changes I suggested in `append_at_cursor` directly, matching other UI toolkits by also handling truncation on paste.